### ISSUE TITLE
Report errors for duplicate case clauses

### ISSUE THREAD

**Issue by DanielRosenwasser** [Microsoft/TypeScript#854]
It is incredibly easy to make a copy/paste error for `case` clauses in a `switch` statement.

```TypeScript
switch (node.kind) {
    // ...
    case SyntaxKind.TupleType:
        // Do something with tuple types.
    case SyntaxKind.TupleType:
        // Do something with union types.
    // ...
}
```

While it is easily arguable that non-exhaustive cases can be allowed in TypeScript, there is no reason not to report duplicated cases.

"Obvious" semantics for `case` clauses in a single `switch` statement are as follows:
* Report when named entities with identical symbols are handled.
* Report when equal string literals are handled.
* Report when equal numeric literals (even 0.0 and 0) are handled.
* Report when `null` is handled more than once.
* Report when `undefined` is handled more than once.

_______________

~~One interesting question is whether exhaustive cases for an enum should report an error when there is a `default` clause.~~

**Comment by danquirk** [comment]
This will tracked/implemented as part of https://github.com/Microsoft/TypeScript/issues/2854 now rather than potentially in #274.
